bool stream_data
---
bool success